Source/Evolve/WarpXEvolve.cpp

```diff
@@ -345,8 +345,11 @@ WarpX::OneStep_nosub (Real cur_time)
     if (do_pml && pml_has_particles) CopyJPML();
     if (do_pml && do_pml_j_damping) DampJPML();
 
-    // ApplyExternalFieldExcitation
-    ApplyExternalFieldExcitationOnGrid();
+    if (cur_time == 0._rt) { // at the first time step, make sure to apply the hard source before fields get evolved
+        // ApplyExternalFieldExcitation
+        ApplyExternalFieldExcitationOnGrid("all");
```

Comment:
```suggestion
        ApplyExternalFieldExcitationOnGrid(ExternalFieldType::All);
```

Reply:
and similarly for the other fields

Reply:
And use "ExternalFieldType::AllExternal" (similar comments below)